pennylane/interfaces/execution.py

```diff
@@ -454,8 +535,11 @@ def cost_fn(params, x):
         for tape in tapes:
             params.extend(tape.get_parameters(trainable_only=False))
         interface = qml.math.get_interface(*params)
+    if interface == "jax":

```

Comment:
Previously, we were running this check in two separate places.  Now we just do it once at the beginning.